File: shared/src/main/kotlin/com/august/domain/model/Wine.kt
package com.august.domain.model

import kotlinx.serialization.Serializable

/**
 * 와인 도메인 모델
 * 
 * 백엔드와 프론트엔드에서 공통으로 사용하는 와인 데이터 구조입니다.
 * 
 * @property id 와인 고유 식별자 (UUID)
 * @property name 와인 이름
 * @property countryCode 생산 국가 코드 (예: KR, FR, IT)
 * @property vintage 빈티지 연도
 * @property price 가격 (달러)
 * @property quantity 재고 수량
 */
@Serializable
data class Wine(
    val id: String,
    val name: String,
    val countryCode: String,
    val vintage: Int,
    val price: Double,
    val quantity: Int
)

File: mobile/src/types/wine.ts
/**
 * 모바일 앱용 와인 타입 정의
 * 
 * 공통 타입을 확장하여 모바일 특화 타입을 정의합니다.
 */

// 기본 와인 타입들
export interface Wine {
  id: string;
  name: string;
  country_code: string;
  vintage: number;
  price: number;
  quantity: number;
  vivino_url?: string | null;
  wine_searcher_url?: string | null;
  created_at: string;
  updated_at: string;
}

export interface CreateWineRequest {
  name: string;
  country_code: string;
  vintage: number;
  price: number;
  quantity: number;
  vivino_url?: string | null;
  wine_searcher_url?: string | null;
}

export interface UpdateWineRequest {
  name?: string;
  country_code?: string;
  vintage?: number;
  price?: number;
  quantity?: number;
  vivino_url?: string | null;
  wine_searcher_url?: string | null;
}

export interface WineListResponse {
  wines: Wine[];
  pagination: {
    page: number;
    limit: number;
    total: number;
    total_pages: number;
  };
}

// 모바일 특화 타입들
export interface WineListItem extends Wine {
  // 모바일에서 추가로 필요한 필드들
  isLowStock?: boolean;
  lastUpdated?: string;
}

export interface WineFormData {
  name: string;
  country_code: string;
  vintage: number;
  price: number;
  quantity: number;
  vivino_url?: string;
  wine_searcher_url?: string;
}
